File: app/src/main/java/com/monk/aidldemo/binder/aidlbinder/AIDLBinder.java
package com.monk.aidldemo.binder.aidlbinder;

import android.os.RemoteException;

import com.monk.aidldemo.IAidlInterface;
import com.monk.aidldemo.bean.Person;

import java.util.List;

/**
 * AIDL 实现方式
 *
 * IAidlInterface 接口的本地实现类 AIDLBinder
 *
 * @author monk
 * @date 2019/1/12
 */
public class AIDLBinder extends IAidlInterface.Stub {

    /**
     * 创建生成的本地 Binder 对象，实现 AIDL 制定的方法
     */
    private final List<Person> mPersons;

    public AIDLBinder(List<Person> mPersons) {
        this.mPersons = mPersons;
    }

    @Override
    public void addPerson(Person p) throws RemoteException {
        mPersons.add(p);
    }

    @Override
    public List<Person> getPersonList() throws RemoteException {
        return mPersons;
    }
}

File: module_fragment/src/main/java/com/monk/modulefragment/aidl/AIDLBinder.kt
package com.monk.modulefragment.aidl

import android.os.RemoteException
import com.monk.modulefragment.aidl.bean.Person

/**
 * AIDL 实现方式
 *
 * IAidlInterface 接口的本地实现类 AIDLBinder
 *
 *
 * 创建生成的本地 Binder 对象，实现 AIDL 制定的方法
 *
 * @author monk
 * @date 2019/1/12
 */
class AIDLBinder(private val mPersons: MutableList<Person>) : IAidlInterface.Stub() {
    @Throws(RemoteException::class)
    override fun addPerson(p: Person) {
        mPersons.add(p)
    }

    @Throws(RemoteException::class)
    override fun getPersonList(): List<Person> {
        return mPersons
    }
}